File: samples/client/petstore/elixir/test/outer_enum_test.exs
defmodule OuterEnumTest do
  use ExUnit.Case, async: true

  alias OpenapiPetstore.Model.EnumTest

  @valid_json """
  {
    "enum_string_required": "lower",
    "enum_string": "UPPER",
    "enum_number": 1.1,
    "outerEnum": "placed",
    "outerEnumInteger": 1
  }
  """

  @tag timeout: :infinity
  test "json_decode/2 with valid JSON" do
    {:ok, enum_test} =
      JSON.decode!(@valid_json)
      |> EnumTest.from_params

    assert enum_test ==
      %EnumTest{
        enum_string_required: "lower",
        enum_string: "UPPER",
        enum_number: 1.1,
        outerEnum: "placed",
        outerEnumInteger: 1
      }
  end
end

File: samples/server/petstore/jaxrs-spec-enum/src/test/java/org/openapitools/model/OuterEnumTest.java
package org.openapitools.model;

import org.junit.Test;

import static org.junit.Assert.assertSame;

public class OuterEnumTest {

    @Test
    public void fromValueUsesCaseInsensitiveMatching() {
        assertSame(OuterEnum.PLACED, OuterEnum.fromValue("PLACED"));
        assertSame(OuterEnum.APPROVED, OuterEnum.fromValue("ApPrOvEd"));
        assertSame(OuterEnum.DELIVERED, OuterEnum.fromValue("delivered"));
    }
}
